File: java/person.java
class Person {
	private String firstName;
	private String lastName;
	private Date dateOfBirth;
	private String phone;
	private ArrayList<Address> address;

	Person(String first, String last, Date dob, String phone, ArrayList<Address> address) {
		this.firstName = first;
		this.lastName= last;
		this.dateOfBirth = dob;
		this.phone = phone;
		this.addresses = [];

		if (address instanceof Address) {
			this.addresses.add(address);
		} else if (address instanceof ArrayList<>) {
			for (int i = 0; i < address.length(); i++) {
				if (!(address[i] instanceof Address)) {
					throw new Exception("Invalid Address...");
				}

				this.addresses = address;
			}
		} else {
			throw new Exception("Invalid Address ...");
		}
	}
	
	public void addAddress(ArrayList<Address> address) {
		if (!(address instanceof Address)) {
			throw new Exception("Invalid Address ... ");
		}
			
		this.addresses.add(address);
	}
}

File: javascript/person.js
const Address = require('./address');

class Person {
	constructor(first, last, dob, phone, address) {
		this.firstName = first;
		this.lastName = last;
		this.dateOfBirth = dob;
		this.phone = phone;
		this.addresses = [];

		if (address instanceof Address) {
			this.addresses.push(address);
		} else if (address.isArray()) {
			address.forEach(item => {
				if (!(item instanceof Address)) {
					throw Error('Invalid Address ...');
				}
				this.addresses = address;
			});
		} else {
			throw Error('Invalid address...');
		}
	}

	addAddress(address) {
		if (!(address instanceof Address)) {
			throw Error('Invalid address..');
		}
		this.addresses.push(address);
	}
}

module.exports = Person;